src/components/data/row/CurrencyRow.js

```diff
@@ -41,81 +35,22 @@ const CurrencyRowComponent = (props: Props) => {
   // Balance stuff:
   const showBalance = useSelector(state => state.ui.settings.isAccountBalanceVisible)
   const balance = useWalletBalance(wallet, tokenId)
+  const icon = <CryptoIcon sizeRem={2} tokenId={tokenId} walletId={wallet.id} />
+  const tickerText = showRate && wallet != null ? <TickerText wallet={wallet} tokenId={tokenId} /> : null
+  const cryptoText = showBalance ? <CryptoText wallet={wallet} tokenId={tokenId} nativeAmount={balance} withSymbol /> : null
+  const fiatText = showBalance ? <FiatText nativeCryptoAmount={balance} tokenId={tokenId} wallet={wallet} /> : null
 
   return (
-    <View style={[styles.container, margin]}>
-      <CryptoIcon sizeRem={2} tokenId={tokenId} walletId={wallet.id} />
-      <View style={styles.nameColumn}>
-        <View style={styles.currencyRow}>
-          <EdgeText style={styles.currencyText}>{currencyCode}</EdgeText>
-          {showRate && wallet != null ? (
-            <EdgeText style={styles.exchangeRateText}>
-              <TickerText wallet={wallet} tokenId={tokenId} />
-            </EdgeText>
-          ) : null}
-        </View>
-        <EdgeText style={styles.nameText}>{name}</EdgeText>
-      </View>
-      {showBalance ? (
-        <View style={styles.balanceColumn}>
-          <EdgeText>
-            <CryptoText wallet={wallet} tokenId={tokenId} nativeAmount={balance} withSymbol />
-          </EdgeText>
-          <EdgeText style={styles.fiatBalanceText}>
-            <FiatText nativeCryptoAmount={balance} tokenId={tokenId} wallet={wallet} />
-          </EdgeText>
-        </View>
-      ) : null}
-    </View>
+    <IconDataRow
+      icon={icon}
+      leftText={currencyCode}
+      leftTextExtended={tickerText}
+      leftSubtext={name}
+      rightText={cryptoText}
+      rightSubText={fiatText}
+      marginRem={marginRem}
+    />
```

Comment:
Oh I see. Why do a "partially factor out" and then a "full factor out" commit separate? "Factor out" should include the whole thing in one commit, this way it can be reviewed to be correct on its own.

Reply:
I wanted to separate the renaming. Figured it would be easier to determine sameness if I didn't factor it out and rename everything generically at the same time.